be/src/olap/rowset/rowset_writer.h

```diff
@@ -107,10 +108,12 @@ class RowsetWriter {
         }
         std::string segment_prefix {InvertedIndexDescriptor::get_index_file_path_prefix(
                 _context.segment_path(segment_id))};
+        bool can_use_ram_dir = _context.compaction_type == ReaderType::READER_BASE_COMPACTION &&
+                               config::inverted_index_ram_dir_enable_when_compaction;
```

Comment:
when config::inverted_index_ram_dir_enable_when_compaction is false then can_use_ram_dir always false this will ignore compaction_type?